common.blocks/modal/_theme/modal_theme_islands.css

```diff
@@ -3,3 +3,52 @@
     border: 1px solid #fff;
     background: #f3f1e7;
 }
+
+@media screen and (max-width: 773px) {
+    .modal_theme_islands .modal__content {
+        box-sizing: border-box;
+        padding: 10px;
+        margin: 0;
+        position: fixed;
+        left: 0;
+        right: 0;
+        bottom: 0;
+        top: 0;
+        overflow: hidden;
+    }
+
+    .input_theme_islands.input_size_xl {
```

Comment:
if this and the next one rules are common, please, move them to separate files for each block or if they are specific to `modal` — add `.modal` to selector.